psyneulink/core/components/functions/nonstateful/optimizationfunctions.py

```diff
@@ -1249,13 +1263,34 @@ def reset(self, default_variable=None, objective_function=None, context=None, **
 
         # Differentiate objective_function using autograd.grad()
         if objective_function is not None and not self.gradient_function:
+
+            if torch is None:
+                raise ValueError("PyTorch is not installed. Please install PyTorch to use GradientOptimization without "
+                                 "specifying a gradient_function.")
+
+            if grad is None:
```

Comment:
`if not 'grad' in dir(torch.func):`
should work as well

Reply:
Changed